src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -452,7 +452,10 @@ def extract_query_filters(self, cmd):
         return FilterSpec(mds_ranks, client_id, client_ip)
 
     def get_perf_data(self, cmd):
-        filter_spec = self.extract_query_filters(cmd)
+        try:
```

Comment:
While you are here, could you also fixup `extract_client_id_from_spec` and `extract_client_ip_from_spec` to check for valid IDs and IPs?

Also, include a test case for the checks.

Reply:
Added checks for valid IDs and IPs.

Reply:
Please add test - qa/tasks/cephfs/test_mds_metrics.py

Reply:
Done.
Added positive & negative test cases

Reply:
Added positive and negative tests

Reply:
- Positive Test: Execute 'ceph fs perf stats' with filter --client_ip.
- Negative Test: Verify exception handling and errno for invalid mds_rank, client_id and client_ip.